sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java

```diff
@@ -396,6 +392,11 @@ public AutoConfiguredOpenTelemetrySdk build() {
         openTelemetrySdk = sdkBuilder.build();
       }
 
+      // NOTE: Shutdown hook registration is untested. Modify with caution.
+      if (registerShutdownHook) {
+        Runtime.getRuntime().addShutdownHook(new Thread(openTelemetrySdk::close));
+      }
```

Comment:
This is tricky to test and IMO not worth it. There's a `boolean Runtime.getRuntime.removeShutdownHook(Thread)` method that returns true if the thread was a registered as a shutdown hook and removed, but that requires complicated coordination for the test to get access to the Thread executing the shutdown operation.

Reply:
You could perhaps pass a `Function<OpenTelemetrySdk, Thread>` via a package private builder method, and verify the SDK passed

Reply:
Turns out that the test ends up being pretty clean if we just split out a package private method int he builder.

Added!